File: openapi-generator/dart/lib/model/tag_modify_add.dart
part of openapi.api;

class TagModifyAdd {
  /* The assets to be added to this tag. */
  List<TaggedAssetBase> assets = [];
  /* The sensors to be added to this tag. */
  List<TaggedSensorBase> sensors = [];
  /* The vehicles to be added to this tag. */
  List<TaggedVehicleBase> vehicles = [];
  /* The machines to be added to this tag. */
  List<TaggedMachineBase> machines = [];
  /* The drivers to be added to this tag. */
  List<TaggedDriverBase> drivers = [];
  TagModifyAdd();

  @override
  String toString() {
    return 'TagModifyAdd[assets=$assets, sensors=$sensors, vehicles=$vehicles, machines=$machines, drivers=$drivers, ]';
  }

  TagModifyAdd.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['assets'] == null) {
      assets = null;
    } else {
      assets = TaggedAssetBase.listFromJson(json['assets']);
    }
    if (json['sensors'] == null) {
      sensors = null;
    } else {
      sensors = TaggedSensorBase.listFromJson(json['sensors']);
    }
    if (json['vehicles'] == null) {
      vehicles = null;
    } else {
      vehicles = TaggedVehicleBase.listFromJson(json['vehicles']);
    }
    if (json['machines'] == null) {
      machines = null;
    } else {
      machines = TaggedMachineBase.listFromJson(json['machines']);
    }
    if (json['drivers'] == null) {
      drivers = null;
    } else {
      drivers = TaggedDriverBase.listFromJson(json['drivers']);
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'assets': assets,
      'sensors': sensors,
      'vehicles': vehicles,
      'machines': machines,
      'drivers': drivers
    };
  }

  static List<TagModifyAdd> listFromJson(List<dynamic> json) {
    return json == null ? new List<TagModifyAdd>() : json.map((value) => new TagModifyAdd.fromJson(value)).toList();
  }

  static Map<String, TagModifyAdd> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, TagModifyAdd>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new TagModifyAdd.fromJson(value));
    }
    return map;
  }
}

File: swagger-codegen/csharp-dotnet2/src/main/CsharpDotNet2/samsara/Model/TagModifyAdd.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace samsara.Model {

  /// <summary>
  /// Specify devices, etc. that should be added to the tag.
  /// </summary>
  [DataContract]
  public class TagModifyAdd {
    /// <summary>
    /// The assets to be added to this tag.
    /// </summary>
    /// <value>The assets to be added to this tag.</value>
    [DataMember(Name="assets", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "assets")]
    public List<TaggedAssetBase> Assets { get; set; }

    /// <summary>
    /// The drivers to be added to this tag.
    /// </summary>
    /// <value>The drivers to be added to this tag.</value>
    [DataMember(Name="drivers", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "drivers")]
    public List<TaggedDriverBase> Drivers { get; set; }

    /// <summary>
    /// The machines to be added to this tag.
    /// </summary>
    /// <value>The machines to be added to this tag.</value>
    [DataMember(Name="machines", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "machines")]
    public List<TaggedMachineBase> Machines { get; set; }

    /// <summary>
    /// The sensors to be added to this tag.
    /// </summary>
    /// <value>The sensors to be added to this tag.</value>
    [DataMember(Name="sensors", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "sensors")]
    public List<TaggedSensorBase> Sensors { get; set; }

    /// <summary>
    /// The vehicles to be added to this tag.
    /// </summary>
    /// <value>The vehicles to be added to this tag.</value>
    [DataMember(Name="vehicles", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "vehicles")]
    public List<TaggedVehicleBase> Vehicles { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class TagModifyAdd {\n");
      sb.Append("  Assets: ").Append(Assets).Append("\n");
      sb.Append("  Drivers: ").Append(Drivers).Append("\n");
      sb.Append("  Machines: ").Append(Machines).Append("\n");
      sb.Append("  Sensors: ").Append(Sensors).Append("\n");
      sb.Append("  Vehicles: ").Append(Vehicles).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
